diff -ruN release.bak/couchdb/lib/couchdb/erlang/lib/couch/priv/couchspawnkillable release/couchdb/lib/couchdb/erlang/lib/couch/priv/couchspawnkillable
--- release.bak/couchdb/lib/couchdb/erlang/lib/couch/priv/couchspawnkillable	2011-02-05 01:26:00.000000000 -0700
+++ release/couchdb/lib/couchdb/erlang/lib/couch/priv/couchspawnkillable	2011-02-05 01:48:49.000000000 -0700
@@ -1,4 +1,4 @@
-#! /bin/sh -e
+#!/system/bin/sh -e

 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
diff -ruN release.bak/erlang/bin/erl_inetrc release/erlang/bin/erl_inetrc
--- release.bak/erlang/bin/erl_inetrc	1969-12-31 17:00:00.000000000 -0700
+++ release/erlang/bin/erl_inetrc	2011-02-05 01:49:58.000000000 -0700
@@ -0,0 +1,4 @@
+%% Use erlang's internal resolver. Native DNS on android is annoying
+{lookup, [file,dns]}.
+%% TODO: Not this,
+{nameserver, {8,8,8,8}}.
\ No newline at end of file
